core/src/main/scala/org/apache/spark/TaskContext.scala

```diff
@@ -123,7 +123,7 @@ abstract class TaskContext extends Serializable {
    *
    * Exceptions thrown by the listener will result in failure of the task.
    */
-  def addTaskCompletionListener(f: (TaskContext) => Unit): TaskContext = {
+  def addTaskCompletionListener[U](f: (TaskContext) => U): TaskContext = {
```

Comment:
Just curious - why do we need this `U` ? AFAIS everywhere we are calling as `addTaskCompletionListener[Unit]` .. may be I am missing something ..

Reply:
I'm curious as well... It creates a lot of caller-site changes.  Looking to the rest of this PR, it seems that all usages are `addTaskCompletionListener[Unit]`.

Reply:
That is the fixed we have as a workaround. Otherwise it wont compile due to the back we have describe in the doc. There are two options for solving the Unit issue, either add () at the end or do this. Otherwise overloading resolution does not work.

Reply:
@debasishg if you see an other option let me know. Keeping one method for example is not an option btw as we need to support two different uses. One that passes a TaskCompletionListener and one that passes a cclosure that is wrapped with a TaskCompletionListener.